master/internal/task/allocation.go

```diff
@@ -194,6 +194,7 @@ func (a *Allocation) Receive(ctx *actor.Context) error {
 						IsReady:           a.logBasedReadinessPassed,
 						ServiceReadyEvent: &msg,
 					})
+					_ = a.db.UpdateAllocationState(a.req.AllocationID, a.state, a.logBasedReadinessPassed)
```

Comment:
have to handle the error here.. at least log it, but with the way this is, if it is failed we are in trouble since the webui will never know to follow the proxy. also i'm not sure is_ready should be a bit in the db, since it only has meaning when the allocation is in memory/active. i think it probably makes more sense for us to just `ctx.Ask()` the actor if it is ready. but let me think about it some.

Reply:
let's just chat over zoom, i think this is probably fine and it doesn't rely on the actor system. but i want to at least think out loud through the other way to write it.